### PR TITLE
add extra composite purchase order param to use status for item

### PR DESCRIPTION
**Pull request by azizbekxm**
## Purpose
- when fetching purchase order there is issue with not found exception in case of member tenant context

## Approach
- added extra composite purchase order param

### CODE REVIEW

File: src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java [hunk 1]
```diff
@@ -512,7 +517,17 @@ protected Future<Void> recreateItemRecords(Map<String, List<Piece>> piecesGroupe
           var dstConfig = InventoryUtils.constructItemRecreateConfig(checkInPiece.getReceivingTenantId(), requestContext, false);
           if (InventoryUtils.allowItemRecreate(srcConfig.tenantId(), dstConfig.tenantId())) {
             logger.info("recreateItemRecords:: recreating item by id '{}', srcTenantId: '{}', dstTenantId: '{}'", piece.getItemId(), srcConfig.tenantId(), dstConfig.tenantId());
-            futures.add(itemRecreateInventoryService.recreateItemInDestinationTenant(itemToRecreate.getCompositePoLine(), piece, srcConfig.context(), dstConfig.context()));
+
+            // Fetch purchase order asynchronously
+            Future<CompositePurchaseOrder> poFuture = purchaseOrderStorageService.getPurchaseOrderByIdAsJson(itemToRecreate.getPoLineId(), requestContext)
```
Comment (BKadirkhodjaev) [inline src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java:522]: I suggest you change the Reference type and variable name to `var purchaseOrderFuture = ...`

File: src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java [hunk 2]
```diff
@@ -512,7 +517,17 @@ protected Future<Void> recreateItemRecords(Map<String, List<Piece>> piecesGroupe
...
+              .map(HelperUtils::convertToCompositePurchaseOrder)
+              .recover(throwable -> {
+                return Future.succeededFuture(null); // Return null to continue processing other requests
+              });
+
+            Future<String> itemFuture = poFuture.compose(purchaseOrder ->
```
Comment (BKadirkhodjaev) [inline src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java:528]: Similar renaming can apply

File: src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java [hunk 3]
```diff
@@ -512,7 +517,17 @@ protected Future<Void> recreateItemRecords(Map<String, List<Piece>> piecesGroupe
...
+              itemRecreateInventoryService.recreateItemInDestinationTenant(purchaseOrder, itemToRecreate.getCompositePoLine(), piece, srcConfig.context(), dstConfig.context()));
+            futures.add(itemFuture);
```
Comment (BKadirkhodjaev) [inline src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java:530]: We can maybe name this as `itemIdFutures`

File: src/main/java/org/folio/orders/utils/PoLineCommonUtil.java [hunk 4]
```diff
@@ -290,4 +290,8 @@ private static void checkFieldsChanged(JsonPathParser oldObject, JsonPathParser
       throw new HttpException(400, error, parameters);
     }
   }
+
+  public static CompositePoLine getPoLine(CompositePurchaseOrder compositePurchaseOrder) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/orders/utils/PoLineCommonUtil.java:294]: need to delete this method from utils. It should not be commonly usagable. If in some places we need to get first po line - it is exceptional case, but not normal situation

File: src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java (hunk 1 above)
Comment (SerhiiNosko) [inline src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java:522]: purchaseOrderStorageService already has method getCompositeOrderByPoLineId() that returns CompositePurchaseOrder

File: src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java [hunk 5]
```diff
@@ -512,7 +517,17 @@ protected Future<Void> recreateItemRecords(Map<String, List<Piece>> piecesGroupe
           var dstConfig = InventoryUtils.constructItemRecreateConfig(checkInPiece.getReceivingTenantId(), requestContext, false);
           if (InventoryUtils.allowItemRecreate(srcConfig.tenantId(), dstConfig.tenantId())) {
             logger.info("recreateItemRecords:: recreating item by id '{}', srcTenantId: '{}', dstTenantId: '{}'", piece.getItemId(), srcConfig.tenantId(), dstConfig.tenantId());
-            futures.add(itemRecreateInventoryService.recreateItemInDestinationTenant(itemToRecreate.getCompositePoLine(), piece, srcConfig.context(), dstConfig.context()));
+
+            // Fetch purchase order asynchronously
+            Future<CompositePurchaseOrder> poFuture = purchaseOrderStorageService.getPurchaseOrderByIdAsJson(itemToRecreate.getPoLineId(), requestContext)
+              .map(HelperUtils::convertToCompositePurchaseOrder)
+              .recover(throwable -> {
+                return Future.succeededFuture(null); // Return null to continue processing other requests
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java:525]: we don't have logs here, so it is not possible to find a reaons here. Alos this change does not relate to current story, we can make it if confident in this

File: src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java [hunk 6]
```diff
@@ -512,7 +517,20 @@ protected Future<Void> recreateItemRecords(Map<String, List<Piece>> piecesGroupe
...
           if (InventoryUtils.allowItemRecreate(srcConfig.tenantId(), dstConfig.tenantId())) {
             logger.info("recreateItemRecords:: recreating item by id '{}', srcTenantId: '{}', dstTenantId: '{}'", piece.getItemId(), srcConfig.tenantId(), dstConfig.tenantId());
-            futures.add(itemRecreateInventoryService.recreateItemInDestinationTenant(itemToRecreate.getCompositePoLine(), piece, srcConfig.context(), dstConfig.context()));
+
+            // Fetch purchase order asynchronously
+            Future<CompositePurchaseOrder> compositePurchaseOrderFuture = purchaseOrderStorageService
+              .getCompositeOrderByPoLineId(itemToRecreate.getPoLineId(), requestContext)
+              .recover(throwable -> {
+                logger.warn("recreateItemRecords:: Composite Purchase order can't be found for poLine: {}", itemToRecreate.getPoLineId());
+                return Future.succeededFuture(null);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java:526]: formatting with 2 lines should be used( these 2 lines are inside of .recover block)

File: src/test/java/org/folio/rest/impl/CheckinReceivingApiTest.java [hunk 7]
```diff
@@ -695,14 +696,15 @@ void testPostCheckInPhysicalFullyReceivedWithChangedLocation() {
 
   private static Stream<Arguments> testPostCheckInPhysicalFullyReceivedEcsArgs() {
     return Stream.of(
-      Arguments.of(12, "checkin-fully-receive-physical-resource-ecs-single.json", 1, 4),
-      Arguments.of(13, "checkin-fully-receive-physical-resource-ecs-multiple.json", 2, 7)
+      Arguments.of(12, "checkin-fully-receive-physical-resource-ecs-single.json", 1, 4, 1),
```
Comment (BKadirkhodjaev) [inline src/test/java/org/folio/rest/impl/CheckinReceivingApiTest.java:699]: Nice optimization

**Review by BKadirkhodjaev — APPROVED** [review]

**Pull request merged**